android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1943,21 +1945,31 @@ private void onLocationPermissionsResult(java.util.Map<String, Boolean> permissi
 
     // Check if fine location permission is granted.
     if (LocationUtils.checkFineLocationPermission(this)) {
-      Logger.i(LOCATION_TAG, "Precise Location Permission Granted");
+      Logger.i(LOCATION_TAG, "Precise location permission granted");
     } else if (LocationUtils.checkCoarseLocationPermission(this)) {
-      Logger.w(LOCATION_TAG, "Only Approximate Location Permission Granted");
-      // Show warning if only coarse location is granted
-      new MaterialAlertDialogBuilder(this, R.style.MwmTheme_AlertDialog)
-          .setTitle("⚠ "+ getString(R.string.limited_accuracy))
-          .setMessage(R.string.precise_location_is_disabled_long_text)
-          .setPositiveButton(R.string.go_to_settings, (dialog, which) -> {
-            Utils.openUri(this, Uri.parse("package:" + getPackageName()), R.string.uri_open_location_failed, Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
-          })
-          .setNegativeButton(R.string.ignore, (dialog, which) -> dialog.dismiss())
-          .setCancelable(false)
-          .show();
+      Logger.w(LOCATION_TAG, "Only approximate location permission granted");
+      if (mToastCount < MAX_TOAST_COUNT_BEFORE_DIALOG) {
```

Comment:
I think let's make it both simpler and less nagging: show the dialog only once after app start/restart.
Then you'll need to keep one bool flag var only.